### ISSUE TITLE
Issue with updating module that uses distributed module

### ISSUE THREAD

**Issue by fisherc2** [timholy/Revise.jl#159]
Hi-

I originally posted this issue on [discourse](https://discourse.julialang.org/t/strange-behavior-in-revise-jl/13531), but never received a response. It appears to be a bug as far as I can tell. The main problem is that I cannot update a module that uses a module that was distributed to other cores.  Below is a minimum working example and my system information.

```
module myModule
    export f
    f() = "I am from myModule."
    #f() = "I can be edited."
end


```
```
module otherModule
    export g
    using myModule
    g() = "I am from other Module."
    #g() = "I cannot be edited."
end

```
```
Top level code:

path = "/home/dfish/Projects/myFolder"
using Distributed
addprocs(4)
@everywhere push!(LOAD_PATH,$path)
@everywhere using myModule,Revise
using otherModule

```
You will see that it is possible to update myModule, but not otherModule. 

System information:

```
versioninfo()
Julia Version 0.7.0
Commit a4cb80f3ed (2018-08-08 06:46 UTC)
Platform Info:
  OS: macOS (x86_64-apple-darwin14.5.0)
  CPU: Intel(R) Xeon(R) CPU E5-1680 v2 @ 3.00GHz
  WORD_SIZE: 64
  LIBM: libopenlibm
  LLVM: libLLVM-6.0.0 (ORCJIT, ivybridge)
Environment:
  JULIA_EDITOR = atom -a
  JULIA_NUM_THREADS = 8
  JULIA_CMDSTAN_HOME = /Users/christopher.fisher/Documents/cmdstan-2.17.1
```

```
 Status `~/.julia/environments/v0.7/Project.toml`
  [c52e3926] Atom v0.7.3
  [336ed68f] CSV v0.3.0
  [593b3428] CmdStan v4.1.1
  [35d6a980] ColorSchemes v1.5.1
  [a93c6f00] DataFrames v0.13.0
  [aaf54ef3] DistributedArrays v0.5.1
  [31c24e10] Distributions v0.16.2
  [28b8d3ca] GR v0.32.3
  [c91e804a] Gadfly v0.7.0
  [682c06a0] JSON v0.19.0
  [e5e0dc1b] Juno v0.5.2
  [b964fa9f] LaTeXStrings v1.0.2
  [429524aa] Optim v0.16.0
  [f0f68f2c] PlotlyJS v0.10.2
  [91a5bcdd] Plots v0.19.2
  [438e738f] PyCall v1.17.1
  [d330b81b] PyPlot v2.6.0
  [1a8c2f83] Query v0.9.3
  [c4c386cf] Rsvg v0.2.1
  [682df890] Stan v3.5.0
  [10a637b5] StanDataFrames v0.0.0 [`~/.julia/dev/StanDataFrames`]
  [60ddc479] StatPlots v0.8.0
  [2913bbd2] StatsBase v0.25.0
```

**Comment by timholy** [comment]
Works for me with `@everywhere using otherModule`. I didn't try it without the `@everywhere`, but I don't see why one should.

**Comment by fisherc2** [comment]
Interesting. It works on my work computer which is a Mac. Are you using a Mac as well?

**Comment by fisherc2** [comment]
I tried again on my personal machine.  I restarted the REPL within Atom and was able to reproduce the behavior. Once I completely shutdown Atom, I was unable to reproduce it. I'm not sure why Atom was interfering with Revise. It doesn't make sense, but it seems to have happened. Things haven't been playing well with each other while the ecosystem has been in flux. I will close this issue since it seems to be working correctly.

**Comment by fisherc2** [comment]
In case it is relevant, I did have a few  intermittent errors along the lines of 

`┌ Warning: The call to compilecache failed to create a usable precompiled cache file for ... `

around the time I was experiencing the issue. I restarted the REPL and thought it may have resolved itself. I'm not sure exactly how Revise works, but I wanted to document that just in case it is related and others experience this issue.
